### DISCUSSION TITLE
Azure AD identifier_key not found

### DISCUSSION THREAD

**Discussion by tobiso95** [directus/directus#11653] [Archive]
Hi, im trying to setup SSO with Azure AD. 
This is my configuration: 
```
      AUTH_PROVIDERS: "microsoft"
      AUTH_MICROSOFT_DRIVER: "openid"
      AUTH_MICROSOFT_CLIENT_ID: "<client-id>"
      AUTH_MICROSOFT_CLIENT_SECRET: "<client-secret>"
      AUTH_MICROSOFT_SCOPE: "openid profile"
      AUTH_MICROSOFT_IDENTIFIER_KEY: "oid"
      AUTH_MICROSOFT_ALLOW_PUBLIC_REGISTRATION: "true"
      AUTH_MICROSOFT_ISSUER_URL: "https://login.microsoftonline.com/<tenant-id>/v2.0/.well-known/openid-configuration"
```

The login seems to work fine, but directus fails to create the user with the following error: `[OpenID] Failed to find user identifier for provider "microsoft"`.

If I change the `IDENTIFIER_KEY` to `sub` it works, all other keys ( like `email` or `oid`) result in this error. Sadly `sub` is not really an option as this is a generated value (how is not disclosed by microsoft) that can not be used to query user data from the [microsoft graph api](https://developer.microsoft.com/en-us/graph). 
I am writing an extension to import user data from azure ad to directus and therefore need the ObjectID (`oid`) of the user.  
I have fetched a token from azure ad manually via postman and can confirm that the 'oid' value is in fact present.

**Comment by aidenfoxx** [comment]
This indicates that the userinfo response from your provider doesn't contain a value for "oid". Is is possible you have to specify an additional claim in the scope to get the value your require?

EDIT: Looking into the docs, it seems MS doesn't provide "oid" in the userinfo response https://docs.microsoft.com/en-us/azure/active-directory/develop/userinfo

This should be solved by the following change https://github.com/directus/directus/pull/11656

**Reply by tobiso95** [reply]
I was not aware of the userinfo endpoint, i thought the id token is used for that. I really have to learn more about the OIDC protocol 😄 
Thanks for the quick response and the change!
